rules/opentitan_test.bzl

```diff
@@ -238,141 +286,87 @@ def opentitan_functest(
 
     all_tests = []
 
-    if "dv" in targets:
-        # Set default DV sim parameters if none are provided.
-        if dv == None:
-            dv = dv_params()
-
-        test_name = "dv_{}".format(name)
-
-        # Regardless if the test is signed or not, DV sims backdoor load flash
-        # with the scrambled VMEM (unless the bootstrap path is used below).
-        test_bin = "{}_prog_sim_dv_scr_vmem64".format(name)
-
-        # If the (flash) test image is to be loaded via bootstrap, then we need
-        # to use the VMEM image that has been split into SPI flash frames.
-        if dv.pop("bootstrap_sw"):
+    target_params = {
+        "sim_dv": dv_params() if (dv == None) else dv,
+        "sim_verilator": verilator_params() if (verilator == None) else verilator,
+        "fpga_cw310": cw310_params() if (cw310 == None) else cw310,
+    }
+
+    for target, params in target_params.items():
+        if target.split("_")[-1] in targets:
+            # Set test name.
+            test_name = "{}_{}".format(target, name)
+            if "manual" not in params.get("tags"):
+                all_tests.append(test_name)
+
+            # Set flash image.
+            flash = "{}_prog_{}_scr_vmem64".format(name, target)
+            if signed:
+                flash = "{}_prog_{}_scr_vmem64_signed_{}".format(
+                    name,
+                    target,
+                    key,
+                )
+
+            # If the (flash) test image is to be loaded via bootstrap in the DV
+            # simulation environment, then we need to use a special VMEM image
+            # that has been split into SPI flash frames. Currently, signed
+            # images loaded via bootstrap in DV sim are not supported.
+            # TODO: support signed bootstap images in DV sim.
+            if target == "sim_dv" and params.pop("bootstrap_sw"):
+                if test_in_rom:
+                    fail("A test runs in ROM cannot be bootstrapped.")
```

Comment:
nit: "Tests that run in ROM cannot be bootstrapped." or "This test runs in ROM and cannot be bootstrapped." or "Test runs in rom and cannot be bootstrapped"

Reply:
Fixed.